code_challenges.rb

```diff
@@ -1,33 +1,70 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
 # MINASWAN
 
-# --------------------1) Create a method that takes in a number and determines if the number is even or odd. Use the test variables provided.
+# --------------------1) Create a method that takes in a number and determines if the number 
+# is even or odd. Use the test variables provided.
 
-num1 = 7
-# Expected output: '7 is odd'
-num2 = 42
-# Expected output: '42 is even'
-num3 = 221
-# Expected output: '221 is odd'
+# num1 = 7
+# # Expected output: '7 is odd'
+# num2 = 42
+# # Expected output: '42 is even'
+# num3 = 221
+# # Expected output: '221 is odd'
 
 
+  if num1.odd? == true
+    puts " #{num1} is very odd "
+  else
+    puts " #{num1} is steady even"
+  
+  end
+  
+  # output =  7 is very odd
+  
+  if num2.odd? == true
+    puts " #{num2} is pretty odd "
+  else
+    puts " #{num2} is adequately even"
 
-# -------------------2) Create a method that takes in a string and removes all the vowels from the string. Use the test variables provided. HINT: Check out this resource: https://ruby-doc.org/core-2.6/String.html#method-i-delete
+  end
 
-album1 = 'Rubber Soul'
-# Expected output: 'Rbbr Sl'
-album2 = 'Sgt Pepper'
-# Expected output: 'Sgt Pppr'
-album3 = 'Abbey Road'
-# Expected output: 'bby Rd'
+  # output = 42 is adequately even
+  
+  if num3.odd? == true
+    puts " #{num3} is so odd "
+  else
+    puts " #{num3} is quite even"
+  
+  end
+# output = 221 is so odd
 
 
 
-# -------------------3) Create a method that takes in a string and checks if the string is a palindrome. A palindrome is the same word spelled forwards or backwards. Use the test variables provided.
+# # -------------------2) Create a method that takes in a string and removes all the vowels from the string. Use the test variables provided. HINT: Check out this resource: https://ruby-doc.org/core-2.6/String.html#method-i-delete
 
-is_palindrome1 = 'Racecar'
-# Expected output: 'Racecar is a palindrome'
-is_palindrome2 = 'LEARN'
-# Expected output: 'LEARN is not a palindrome'
-is_palindrome3 = 'Rotator'
-# Expected output: 'Rotator is a palindrome'
+
+ album1 = 'Rubber Soul'
+# # # Expected output: 'Rbbr Sl'
+ album2 = 'Sgt Pepper'
+# # # Expected output: 'Sgt Pppr'
+ album3 = 'Abbey Road'
+# # # Expected output: 'bby Rd'
+#
+p album1.delete ("u" "o")
+p album2.delete ("e")
+p album3.delete ("a" "e" "o")
```

Comment:
So this is a good example of accomplishing the end result without building the unique code that can be called upon as many times as you like. 

So the question is almost never "delete the vowels from this one album name" although it looks like that. The question is always "give ANY album name delete all the vowels" 

so using your code I can set up some steps 

```ruby
def vowel_delete album
   
end
``` 
1. the def and end are the surrounding parts to any ruby method
2. The name of the method describes what it will do "delete the vowels" 
3. and it takes in an argument of an album
4. add in YOUR logic (plus a bit extra)
```ruby
def vowel_delete album
    album.delete('a','e','i','o','u','A','E','I','O','U')
end
``` 
 Now I can call on them all without any extra code
 ```ruby
p vowel_delete(album1)
p vowel_delete(album2)
p vowel_delete(album3)
p vowel_delete("White Album")
p vowel_delete("Let it be")
p vowel_delete("Magical Mystery Tour")
```
...etc

Reply:
Yeah when we went over the assessment I had a couple of ah ha moments.